lib/index.js

```diff
@@ -2,115 +2,113 @@
 
   var scheme = {
-    authenticate: function (request, reply) {
+    authenticate: async function (request, h) {
 
       var req = request.raw.req;
       var authorization = req.headers.authorization;
       if (!authorization) {
-        return reply(Boom.unauthorized(null, 'Bearer'));
+        throw Boom.unauthorized(null, 'Bearer');
       }
 
       var parts = authorization.split(/\s+/);
 
       if (parts.length !== 2) {
-        return reply(Boom.badRequest('Bad HTTP authentication header format', 'Bearer'));
+        throw Boom.badRequest('Bad HTTP authentication header format', 'Bearer');
       }
 
       if (parts[0].toLowerCase() !== 'bearer') {
-        return reply(Boom.unauthorized(null, 'Bearer'));
+        throw Boom.unauthorized(null, 'Bearer');
       }
 
-      if(parts[1].split('.').length !== 3) {
-        return reply(Boom.badRequest('Bad HTTP authentication header format', 'Bearer'));
+      if (parts[1].split('.').length !== 3) {
+        throw Boom.badRequest('Bad HTTP authentication header format', 'Bearer');
       }
 
       var token = parts[1];
 
       var getKey = isFunction(settings.key) ?
         settings.key :
-        function(req, token, callback) { callback(null, settings.key); };
+        function (req, token) { return { key: settings.key }; };
 
-      getKey(request, token, function(err, key, extraInfo){
-        if (err) { return reply(Boom.wrap(err)); }
-        // handle err
-        jwt.verify(token, key, settings, function(err, decoded) {
-          
-          if(err) {
-            return reply(Boom.unauthorized( 'JSON Web Token validation failed: ' + err.message, 'Bearer'));
-          }
-
-          if (!settings.validateFunc) {
-            return reply.continue({ credentials: decoded });
-          }
-
-          settings.validateFunc(decoded, extraInfo, function (err, isValid, credentials) {
-
-            credentials = credentials || null;
+      let keyResult;
+      try {
+        keyResult = await getKey(request, token);
+      } catch (err) {
+        throw err;
+      }
+      const { key, extraInfo } = keyResult;
 
-            if (err) {
-              return reply(err, null, { credentials: credentials, log: { tags: ['auth', 'jwt'], data: err } });
-            }
+      let decoded;
+      try {
+        decoded = jwt.verify(token, key, settings);
+      } catch (err) {
+        throw Boom.unauthorized('JSON Web Token validation failed: ' + err.message, 'Bearer');
+      }
 
-            if (!isValid) {
-              return reply(Boom.unauthorized('Invalid token', 'Bearer'), null, { credentials: credentials });
-            }
+      if (!settings.validateFunc) {
+        return h.authenticated({ credentials: decoded });
+      }
 
-            if (!credentials || typeof credentials !== 'object') {
+      let validateResult;
+      try {
+        validateResult = await settings.validateFunc(decoded, extraInfo);
+      } catch (err) {
+        throw err;
```

Comment:
This is the same as not having the try catch

Reply:
Also are you sure that jus throwing the same error is equivalent to ` return reply(err, null, { credentials: credentials, log: { tags: ['auth', 'jwt'], data: err } });`?